app/invocation/invocation_action_card.tsx

```diff
@@ -196,6 +204,36 @@ export default class InvocationActionCardComponent extends React.Component<Props
     );
   }
 
+  isFileNode(node: Node): node is build.bazel.remote.execution.v2.IFileNode {
+    return node.hasOwnProperty("isExecutable");
+  }
+
+  handleFileClicked(node: any) {
+    let digestString = node.digest.hash + "/" + node.digest.sizeBytes;
+    let dirUrl = "bytestream://" + this.getCacheAddress() + "/blobs/" + digestString;
+
+    if (this.state.treeShaToExpanded.get(digestString)) {
+      this.state.treeShaToExpanded.set(digestString, false);
+      this.forceUpdate();
+      return;
+    }
+    if (this.isFileNode(node)) {
+      rpcService.downloadBytestreamFile(node.name, dirUrl, this.props.model.getId());
+      return;
+    }
+    rpcService
+      .fetchBytestreamFile(dirUrl, this.props.model.getId(), "arraybuffer")
+      .then((dirBuff: any) => {
+        let tempDir = build.bazel.remote.execution.v2.Directory.decode(new Uint8Array(dirBuff));
```

Comment:
nit: avoid using "temp" in variable names -- local variables are always temporary so the "temp" prefix isn't adding much value